packages/astro/src/content/data-store.ts

```diff
@@ -0,0 +1,135 @@
+import { promises as fs, type PathLike, existsSync } from 'fs';
+export class DataStore {
+	#collections = new Map<string, Map<string, any>>();
```

Comment:
I've not, because the store is a singleton and could be instantiated in several different ways and then shared. If it does turn out that it's needed then we can work it out, but for now I think it's ok without. Of course when we get to using libsql, we'll definitely need it.